pennylane/devices/default_qubit_autograd.py

```diff
@@ -158,11 +158,18 @@ def _get_unitary_matrix(self, unitary):
             the unitary in the computational basis, or, in the case of a diagonal unitary,
             a 1D array representing the matrix diagonal.
         """
-        op_name = unitary.name
+        op_name = unitary.name.split(".inv")[0]
+
         if op_name in self.parametric_ops:
             if op_name == "MultiRZ":
-                return self.parametric_ops[unitary.name](*unitary.parameters, len(unitary.wires))
-            return self.parametric_ops[unitary.name](*unitary.parameters)
+                mat = self.parametric_ops[op_name](*unitary.parameters, len(unitary.wires))
+            else:
+                mat = self.parametric_ops[op_name](*unitary.parameters)
+
+            if unitary.inverse:
+                mat = self._transpose(self._conj(mat))
+
+            return mat

```

Comment:
If we missed this for all of our default devices, how can we know if it also works for external devices?

Reply:
* The integration tests check that `inv` is applied correctly, so we know all external devices _apply_ `inv() correctly.
* Currently, no external devices use backprop, so this bug can't be affecting any of them.

But yes, in general, if we have more external devices that support backprop, this will be an issue.

Reply:
I guess I don't see how this is only a backprop issue as there's nothing backprop specific in this method. (Pleas correct me if I am wrong here) but looks like it just completely missed inverting the gates at application time. I feel like the integration tests should've caught this regardless of backprop and it worries me that it didn't

Reply:
Ah, so what was happening was that:

* User requests `RX(0.54, wires=0).inv()` (or something like that).

* The gate name was mutated to `RX.inv`

* As a result, `if op_name in self.parametric_ops:` was failing

* `return unitary.eigvals` and `return unitary.matrix` was being called. Both of these take the inverse into account but using vanilla NumPy, hence why the evaluation worked but backprop didn't. Other diff methods such as parameter-shift would also pass.

Reply:
Actually, typing that out gave me an idea; with `qml.math`, could we make `unitary.matrix` and `unitary.eigvals` framework independent? This would remove the entire need for `jax_ops`, `tf_ops`, etc., and this logic here.